cmd/convertor/builder/builder.go

```diff
@@ -363,6 +365,16 @@ func (b *overlaybdBuilder) Build(ctx context.Context) (v1.Descriptor, error) {
 	// when errors are encountered fallback to regular conversion
 	if convertedDesc, err := b.engine.CheckForConvertedManifest(ctx); err == nil && convertedDesc.Digest != "" {
 		logrus.Infof("Image found already converted in registry with digest %s", convertedDesc.Digest)
+		cbuf, err := json.Marshal(convertedDesc)
+		if err != nil {
+			return v1.Descriptor{}, err
+		}
+		// ensure that output tag is pushed even if the manifest is already converted and found in registry
+		// push output tag only if the pusher is tagPusher
+		if err := uploadBytes(ctx, b.pusher, convertedDesc, cbuf); err != nil {
```

Comment:
Have you tested this out? It is my expectation that this will fail to push as you are serializing the descriptor rather than the manifest itself.

Reply:
I was working on this in parallel it seems, so this shows what I mean: https://github.com/containerd/accelerated-container-image/commit/2ac3a125bc4d45d8e8bde2cec6b2eaedf4e9476b.